### PR TITLE
Allow to enable/disable specific interrupt in IRQ controller

### PR DESCRIPTION
**Pull request by HBuczynski**
## Description

New hal API:
 - `hal_interruptsEnable` -> `hal_interruptsEnableAll`
 - `hal_interruptsDisable` -> `hal_interruptsDisableAll`
 - `hal_interruptsEnable(unsigned int irqn)`
 - `hal_interruptsDisable(unsigned int irqn)`

Changes are applied to all architectures and device drivers.

## Motivation and Context

Device drivers disabled all interrupts which is not an optimal method of synchronization. In the new approach, the device disable only specific interrupt.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## How Has This Been Tested?

- [ ] Already covered by automatic testing.
- [ ] New test added: (add PR link here).
- [x] Tested by hand on: (zedboard, ia32-generic-qemu, zynq7000-qemu).

## Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing linter checks and tests passed.
- [ ] My changes generate no new compilation warnings for any of the targets.

## Special treatment

- [ ] This PR needs additional PRs to work (list the PRs, preferably in merge-order).
- [x] I will merge this PR by myself when appropriate.

### CODE REVIEW

File: hal/armv7m/interrupts.c [hunk 1]
```diff
@@ -52,24 +52,40 @@ static void interrupts_nvicSetPriority(s8 irqn, u32 priority)
...
 {
 	__asm__ volatile("cpsid if");
 }
 
 
+__attribute__((section(".noxip"))) void hal_interruptsEnable(unsigned int irqn)
+{
+	if (irqn >= 0x10) {
+		interrupts_nvicSetIRQ(irqn - 0x10, 1);
+	}
```
Comment (agkaminski) [inline hal/armv7m/interrupts.c:71]: You need to handle irq < 16 here too, otherwise disable on SYSTICK won't work (it's an exception with irq number 15)

Comment (agkaminski) [reply hal/armv7m/interrupts.c:71]: Or just disable all interrupts in case of armv7m timers

Comment (HBuczynski) [reply hal/armv7m/interrupts.c]: I have reverted the change for `SYSTICK` in `stm32/l4/timer.c`.

**Review by agkaminski — APPROVED** [review]

**Review by gerard5 — APPROVED** [review]
LGTM , tested on HW.

**Pull request merged**